gremlin-driver/src/test/java/org/apache/tinkerpop/gremlin/driver/WebSocketClientBehaviorIntegrateTest.java

```diff
@@ -88,6 +89,44 @@ public void shutdown() {
         rootLogger.removeAppender(recordingAppender);
     }
 
+    /**
+     * Tests that client is correctly sending user agent during web socket handshake by having the server return
+     * the captured user agent.
+     */
+    @Test
+    public void shouldIncludeUserAgentInHandshakeRequest() {
```

Comment:
Please add the following test scenarios:

1. Validate that the user agent is sent only once per WebSocket connection.

Reply:
That sounds like a good test to have, I will add it.

Reply:
On further inspection I'm less sure that any test of this nature would be useful. Since the user agent is sent as an http header during the handshake, it is by definition only sent once per connection. Once the handshake is done and the connection established, there isn't really anywhere to check for a user agent being sent again.

Reply:
sure, feel free to ignore the suggestion of this new test.